### ISSUE TITLE
prepVGA and VGA parameters

### ISSUE THREAD

**Issue by yeinnoh12** [orange-vertex/rdepthmap#4]
Hi there! Firstly, thank you so much for developing the R depthmap. This was very needed! I'm trying to use this to run spatial analysis on a building.

I have several questions about perpVGA and VGA functions:
- In prepVGA, does the 'gridsize' equal to 'grid spacing' in depthmapX?
- What does the fillx and filly do exactly? I'm setting them as both 10, but I don't see any difference in the result.
- In perpVGA, why does the maxVisibility has a limit of 400? When I run the same analysis in depthmapX, the visibility hits even over 600.
- If I run the analysis with maxVisibility of 400 anyway, it returns an error message. Let me show you the whole syntax that I was running.

> prepVGA('CODA', 'CODA_VGA', 10, 10,  gridSize = 5,  maxVisibility = 400, makeBoundaryGraph = FALSE, makeGraph = TRUE)
VGA('CODA_VGA', 'VGAout', "visibility-global", 40)
CODA_result <- getPointmapData('VGAout', scale = 1, cliPath = getDefaultCLILocation(), verbose = FALSE)

and then I get:

> Error in points2grid(points, tolerance, round) : 
  cannot derive grid parameters from a single point!

Again, thanks a lot for developing this!

**Comment by pklampros** [comment]
Hi @yeinnoh12,

It appears you have discovered a function that was not supposed to be available. While that's how the CLI works (has a catch-all prepVGA function), for the package I split the functionality into four functions for more flexibility: `createGrid`, `fillGrid`, `makeVGAGraph`, `unmakeVGAGraph`. It seems I just forgot to delete the prepVGA function... In fact the reason you have those errors is because the function is wrong and the two parameters gridSize and maxVisibility are reversed (thus you only create one point, thus R can not create a grid from one point). This is also the reason for the maxVisibility limitations, these were gridSize limitations..
fillX and fillY are now in the `fillGrid` function and they are essentially the points inside the space where you want to fill the grid as you would "click" do in depthmap.

Your code should become from:

> prepVGA('CODA', 'CODA_VGA', 10, 10, gridSize = 5, maxVisibility = 400, makeBoundaryGraph = FALSE, makeGraph = TRUE)

to:

> createGrid('CODA', 'CODA_VGA', gridSize = 5)
> fillGrid('CODA_VGA', 10, 10)
> makeVGAGraph('CODA_VGA', maxVisibility = 400, makeBoundaryGraph = FALSE, makeGraph = TRUE)

I will now remove the prepVGA function, but just to let you know you can look at the parameters of all functions by typing (in the console) ?fillGrid or just searching for them in the help

Please try them out and let me know if the problem is fixed

P

**Comment by yeinnoh12** [comment]
Thanks for the response.
I used the lines you sent me and they sent me back an error:

> createGrid('CODA', 'CODA_VGA', gridSize = 5)
Error in depthmapXcli(params, cliPath, verbose) : 
  Either -pp or -pf must be given

> fillGrid('CODA_VGA', 10, 10)
Error in data.frame(x = fillX, y = fillY) : 
  argument "fillY" is missing, with no default

> makeVGAGraph('CODA_VGA', maxVisibility = 400, makeBoundaryGraph = FALSE, makeGraph = TRUE)
Error in makeVGAGraph("CODA_VGA", maxVisibility = 400, makeBoundaryGraph = FALSE,  : 
  unused arguments (makeBoundaryGraph = FALSE, makeGraph = TRUE)

This is why I searched up for depthmapXcli and found your functions.
Is it possible that I'm not entering a parameter here?

Thanks!
Yeinn

**Comment by pklampros** [comment]
Hmmm, which platform are you on? windows? If yes, reinstall the package and try again. This was a problem with the previous versions of depthmapXcli but should be fixed now..!

For the other commands I was just typing out of memory and didn't look into the code.
For most (all?) functions you can give it the name of the input file (first parameter) and the name of the output file (second parameter). If you don't define a second parameter the function will just use the first parameter as the output file and overwrite it.
Thus, for fillGrid it needs to be:
`fillGrid('CODA_VGA', fillX = 10, fillY = 10)`
otherwise it takes the first 10 to mean the name of the output file. 
For makeVGAGraph makeBoundaryGraph needs to be just boundaryGraph and you don't need makeGraph as that's the name of the command:
`makeVGAGraph('CODA_VGA', maxVisibility = 400, boundaryGraph = FALSE)`

**Comment by yeinnoh12** [comment]
reinstalling the package didn't solve the issue of giving me back:
> Error in depthmapXcli(params, cliPath, verbose) : 
  Either -pp or -pf must be given

The other two functions seem to work now, although they are asking to make the grid first.

**Comment by pklampros** [comment]
I can't replicate the issue anymore as I've tried it in two different windows installations and it works...

Try to force re-installation:
```
install_github("orange-vertex/rdepthmap", force = TRUE)
```
and make sure that the output in the console says that it has been installed

**Comment by yeinnoh12** [comment]
Forcing the installation gave me the same issue. However, I found that it returns error messages when I try to install from gitub.

> Error: Failed to install 'rdepthmap' from GitHub:
  (converted from warning) packages ‘rgeos’, ‘rgdal’ are in use and will not be installed

So I installed the required packages 'rgeos' and 'rgdal' beforehand and it works now! I think it was a version issue.

Thanks a lot.

Yeinn
